kubernetes/client.go

```diff
@@ -304,12 +305,39 @@ func NewClientFromConfig(config *rest.Config) (*IstioClient, error) {
 		return nil, err
 	}
 
+	if client.IsOpenShift() {
+		// TODO This one or the project one..?
+		openshiftAPI, err := newClientForAPI(config, OpenshiftGroupVersion, types)
+		if err != nil {
+			return nil, err
+		}
+
+		client.okd = openshiftAPI
+	}
+
 	client.istioConfigApi = istioConfigAPI
 	client.istioNetworkingApi = istioNetworkingAPI
 	client.istioAuthenticationApi = istioAuthenticationAPI
 	client.istioRbacApi = istioRbacApi
 	client.maistraAuthenticationApi = maistraAuthenticationAPI
 	client.maistraRbacApi = maistraRbacApi
+
+	// Init client cache
+	// Note that cache will work only in full permissions scenarios (similar permissions as mixer/istio-telemetry component)
```

Comment:
yeah, no full permissions scenarios need to be revisited how to use them with the cache.
Probably namespaces/project can't be cached, but perhaps workloads and istio objects can get benefits.